cmake/Modules/FindAmgXSolver.cmake

```diff
@@ -0,0 +1,78 @@
+# - Try to find AmgX
+# Once done this will define
+#
+#  AMGXSOLVER_FOUND        - system has PETSc
+#  AMGXSOLVER_INCLUDES     - the PETSc include directories
+#  AMGXSOLVER_LIBRARIES    - Link these to use PETSc
+#
+# Redistribution and use is allowed according to the terms of the BSD license.
+# For details see the accompanying COPYING-CMAKE-SCRIPTS file.
+#
+
+#cmake_policy(VERSION 3.3)
+
+if( NOT PETSC_FOUND )
+  return()
+endif()
+
+set(AMGX_VALID_COMPONENTS
+    C
+    CXX)
+
+SET(CUDA_DIR "$ENV{CUDA_DIR}" CACHE PATH "The path to CUDA.")
+find_package(CUDA)
+
+if(${CUDA_FOUND})
+  SET(CUDA_LIBRARIES ${CUDA_LIBRARIES} ${CUDA_CUBLAS_LIBRARIES} ${CUDA_cusparse_LIBRARY} ${CUDA_cusolver_LIBRARY})
+  #MESSAGE("-- Finding CUDA - Success")
+  message("Cuda = ${CUDA_LIBRARY_DIRS} ${CUDA_LIBRARIES}")
+
+  # search AMGX header
+  find_path(AMGX_INCLUDE_DIR amgx_c.h amgx_capi.h
+    PATHS ${AMGX_ROOT} ${AMGX_DIR}
+    PATH_SUFFIXES include include/amgx
+    NO_DEFAULT_PATH
+    DOC "Include directory of AMGX")
+
+  find_library(AMGX_LIBRARY
+      NAMES "amgx" "amgxsh"
+      PATHS ${AMGX_ROOT}
+      PATH_SUFFIXES "lib" "lib" "lib/${CMAKE_LIBRARY_ARCHITECTURE}"
+      NO_DEFAULT_PATH)
+
+  find_path(AMGX_WRAPPER_INCLUDE_DIR AmgXSolver.hpp
+    PATHS ${AMGX_WRAPPER_ROOT} ${AMGX_WRAPPER_DIR}
+    PATH_SUFFIXES include
+    NO_DEFAULT_PATH
+    DOC "Include directory of AMGX_WRAPPER")
+
+  find_library(AMGX_WRAPPER_LIBRARY
+      NAMES "AmgXWrapper"
+      PATHS ${AMGX_WRAPPER_ROOT}
+      PATH_SUFFIXES "lib" "lib" "lib/${CMAKE_LIBRARY_ARCHITECTURE}"
```

Comment:
stick to modern idioms, ie no ROOT, no specifying suffixes. users can use AmgXSolver_DIR